packages/editor-ui/src/utils/__tests__/pairedItemUtils.test.ts

```diff
@@ -0,0 +1,400 @@
+			const actual = getPairedItemsMapping(MOCK_EXECUTION);
+			const expected = {
+				DebugHelper_r0_o0_i0: new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'If_r0_o0_i0',
+					'Edit Fields_r1_o0_i0',
+					'Edit Fields1_r1_o0_i0',
+				]),
+				DebugHelper_r0_o0_i1: new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'If_r0_o1_i0',
+					'Edit Fields_r0_o0_i0',
+					'Edit Fields1_r0_o0_i0',
+				]),
+				'Edit Fields1_r0_o0_i0': new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'DebugHelper_r0_o0_i1',
+					'If_r0_o1_i0',
+					'Edit Fields_r0_o0_i0',
+				]),
+				'Edit Fields1_r1_o0_i0': new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'DebugHelper_r0_o0_i0',
+					'If_r0_o0_i0',
+					'Edit Fields_r1_o0_i0',
+				]),
+				'Edit Fields_r0_o0_i0': new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'DebugHelper_r0_o0_i1',
+					'If_r0_o1_i0',
+					'Edit Fields1_r0_o0_i0',
+				]),
+				'Edit Fields_r1_o0_i0': new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'DebugHelper_r0_o0_i0',
+					'If_r0_o0_i0',
+					'Edit Fields1_r1_o0_i0',
+				]),
+				If_r0_o0_i0: new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'DebugHelper_r0_o0_i0',
+					'Edit Fields_r1_o0_i0',
+					'Edit Fields1_r1_o0_i0',
+				]),
+				If_r0_o1_i0: new Set([
+					'When clicking "Test workflow"_r0_o0_i0',
+					'DebugHelper_r0_o0_i1',
+					'Edit Fields_r0_o0_i0',
+					'Edit Fields1_r0_o0_i0',
+				]),
+				'When clicking "Test workflow"_r0_o0_i0': new Set([
+					'DebugHelper_r0_o0_i0',
+					'DebugHelper_r0_o0_i1',
+					'If_r0_o0_i0',
+					'If_r0_o1_i0',
+					'Edit Fields_r0_o0_i0',
+					'Edit Fields_r1_o0_i0',
+					'Edit Fields1_r0_o0_i0',
+					'Edit Fields1_r1_o0_i0',
+				]),
+			};
+			expect(actual).toEqual(expected);
+		});
+
+		it('should skip mapping if execution has more than max items overall', () => {
+			const mockExecution: IExecutionResponse = {
+				data: {
+					startData: {},
+					resultData: {
+						runData: {
+							Start: [
+								{
+									startTime: 1706027170005,
+									executionTime: 0,
+									source: [],
+									executionStatus: 'success',
+									data: {
+										main: [[]],
+									},
+								},
+							],
+							DebugHelper: [
+								{
+									startTime: 1706027170005,
+									executionTime: 1,
+									source: [{ previousNode: 'Start' }],
+									executionStatus: 'success',
+									data: {
+										main: [[]],
+									},
+								},
+							],
+						},
+						pinData: {},
+						lastNodeExecuted: 'DebugHelper',
+					},
+					executionData: {
+						contextData: {},
+						nodeExecutionStack: [],
+						metadata: {},
+						waitingExecution: {},
+						waitingExecutionSource: {},
+					},
+				},
+				mode: 'manual',
+				startedAt: new Date('2024-01-23T16:26:10.003Z'),
+				stoppedAt: new Date('2024-01-23T16:26:10.011Z'),
+				status: 'success',
+				finished: true,
+			};
```

Comment:
I think we could use satisfies [`satisfies IExecutionResponse`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html) here. That way we wouldn't need most of the optional chaining`?.`

Reply:
thanks. did not know about this. will try it

Reply:
got rid of the unnecessary optional chaining to simplify here